src/backend/core/cart/models.py

```diff
@@ -174,6 +174,38 @@ class Cart(models.Model):
         ShippingMethodCountry, null=True, on_delete=models.SET_NULL, related_name="+"
     )
 
+    def is_valid(self):
+        """
+        validate cart
+        * check if cart is not empty (has some cart_items)
+        * check if cart_items are instock
+        * check if has shipping_info
+        * check if has billing_info
+        * check if has payment_method_country
+        * check if has shipping_method_country
+        """
+
+        items = self.cart_items.all()
+        if not items or len(items) == 0:
+            raise "Cart is empty"
+        for item in items:
+            if item.product_variant.stock_quantity < 1:
+                raise f"Product {item.product_variant} is out of stock"
```

Comment:
We should also check if each cart item's quantity is <= its stock quantity
(e.g. you cannot buy 4 coffees if there are only 3 on stock)

Reply:
Check it now pls:)